Model/Gateway.php

```diff
@@ -183,6 +188,26 @@ public function tokenPurchase(
         return $this->doRequest("POST", "/purchases", $payload);
     }
 
+    /**
+     * Validate token
+     *
+     * @param string $pmntsToken
+     * @return array
+     * @throws \JsonException
+     */
+    private function prepareGoogleToken(string $pmntsToken)
+    {
+        try {
+            $token = json_decode($pmntsToken, true);
+            if (!is_array($token)) {
+                throw new \JsonException("JSON Syntax error. Invalid Token");
+            }
+        } catch (\Exception $e) {
```

Comment:
Why are you catching the exception you've just thrown, and then throwing the same exception?
`json_decode` won't throw an exception for things like invalid syntax, so what exception is being caught here?